pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1383,6 +1383,77 @@ func parseIpspec(ipspec *zconfig.Ipspec,
 		config.DhcpRange.Start = start
 		config.DhcpRange.End = end
 	}
+
+	// take some default
+	if config.DhcpRange.Start == nil || config.DhcpRange.End == nil ||
+		config.Gateway == nil {
+		addrCount := types.GetIPAddrCountOnSubnet(config.Subnet)
+		if addrCount == 0 {
+			return nil
+		}
+		base := types.GetIPSubnetBase(config.Subnet.IP, config.Gateway)
+		ipStart := 1
+		if base != 0 {
+			ipStart = base
+		}
+		dhcpRangeStart := ipStart + 1
+		if addrCount >= types.LargeAppCount {
+			// when gateway is less than the half point,
+			// the dhcpRange starts at the half point,
+			// provided the DhcpRange.End is not set
```

Comment:
I think the code should assume that End and Start are either both set or neither is set. We only need default settings for cases which exist in deployment and one set and not the other doesn't exist.